src/app/src/main/kotlin/org/vaccineimpact/api/app/ChunkedFileCache.kt

```diff
@@ -29,7 +30,14 @@ class ChunkedFileCache(private val flushInterval: Long = TimeUnit.HOURS.toMillis
     override fun put(item: ChunkedFile)
     {
         recycle()
-        memoryCache[item.uniqueIdentifier] = MutablePair(item, System.currentTimeMillis())
+        if (memoryCache[item.uniqueIdentifier] == null)
+        {
+            memoryCache[item.uniqueIdentifier] = MutablePair(item, System.currentTimeMillis())
+        }
+        else
+        {
+            memoryCache[item.uniqueIdentifier]!!.left.uploadedChunks.putAll(item.uploadedChunks)
```

Comment:
Does this new step correspond to something which would have happened if the memory cache entry hadn't initially evaluated to null (but since it was, has to be repeated here)?

When this method is called by getFileMetadata, having evaluated that the cache entry does not yet exist, it returns the item ChunkedFile which was passed to this function (and which, under the race condition, does not end up in the cache as its equivalent is already there). The caller of getFileMetada, uploadBurdenEstimateFile, sets uploadedChunks in this returned metadata. Is that OK under the race condition because of the extra step taken here?

Reply:
Huh, well spotted. Suggests that this wasn't in fact a fix, and the working behaviour we saw on UAT was just a coincidence!